### PR TITLE
feat(helm): add update strategy on controller deployment

### PR DESCRIPTION
**Pull request by sebglon**
Implement issue #692

### CODE REVIEW

**Review by benashz — COMMENTED** [review]
Thanks for the PR. Please take a look at my suggestions that need to be resolved before merging.

File: chart/values.yaml [hunk 1]
```diff
@@ -9,6 +9,17 @@ controller:
...
+  # Kubernetes supports types Recreate, and RollingUpdate
+  # ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
+  # @type: object
+  strategy: {}
+  # rollingUpdate:
+  #   maxSurge: 1
+  #   maxUnavailable: 0
+  # type: RollingUpdate
+
+
```
[the remark this answers is not in the document]
Comment (sebglon) [reply chart/values.yaml:23]: done

File: chart/values.yaml [hunk 2]
```diff
@@ -9,6 +9,17 @@ controller:
...
 
+  # Configure update strategy for multi-replica deployments.
+  # Kubernetes supports types Recreate, and RollingUpdate
+  # ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
+  # @type: object
+  strategy: {}
+  # rollingUpdate:
+  #   maxSurge: 1
+  #   maxUnavailable: 0
+  # type: RollingUpdate
```
[the remark this answers is not in the document]
Comment (sebglon) [reply chart/values.yaml]: done

**Review by benashz — COMMENTED** [review]
LGTM! Thank you for your contribution to HashiCorp.

**Pull request merged**